src/parser.rs

```diff
@@ -777,163 +872,208 @@ impl<'a> DocParser<'a> {
-                }
-              }
-            }
-            ModuleDecl::ExportDefaultExpr(export_expr) => {
-              if let Expr::Ident(ident) = export_expr.expr.as_ref() {
-                if let Some(doc_node) = symbols.get(&ident.sym.to_string()) {
-                  doc_entries.push(DocNode {
-                    name: String::from("default"),
-                    declaration_kind: DeclarationKind::Export,
-                    ..doc_node.clone()
-                  });
-                }
-              } else if let Some(js_doc) =
-                js_doc_for_range(parsed_source, &export_expr.range())
-              {
-                let location = get_location(parsed_source, export_expr.start());
-                doc_entries.push(DocNode::variable(
-                  String::from("default"),
-                  location,
-                  DeclarationKind::Export,
-                  js_doc,
-                  super::variable::VariableDef {
-                    kind: deno_ast::swc::ast::VarDeclKind::Var,
-                    ts_type: super::ts_type::infer_ts_type_from_expr(
-                      parsed_source,
-                      export_expr.expr.as_ref(),
-                      true,
-                    ),
-                  },
-                ));
+    let is_ambient = exports.is_empty() && !module_has_import(module_symbol);
+    if is_ambient || self.private {
+      let mut handled_symbols =
+        exports.values().map(|n| n.1).collect::<HashSet<_>>();
+      for child_id in module_symbol.child_decls() {
+        if !handled_symbols.insert(child_id) {
+          continue; // already handled
+        }
+        let child_symbol = module_symbol.symbol(child_id).unwrap();
+        for decl in child_symbol.decls() {
+          if let Some(node) = decl.maybe_node() {
+            let is_declared = self.get_declare_for_symbol_node(node);
+            if is_declared || self.private {
+              if let Some(mut doc_node) = self.get_doc_for_symbol_node_ref(
+                module_symbol,
+                child_symbol,
+                node,
+              ) {
+                doc_node.declaration_kind = if is_declared {
+                  DeclarationKind::Declare
+                } else {
+                  debug_assert!(self.private);
+                  DeclarationKind::Private
+                };
+                doc_nodes.push(doc_node);
               }
             }
-            _ => {}
           }
         }
       }
     }
 
-    if is_ambient {
-      doc_entries.extend(ambient_entries);
+    doc_nodes
+  }
+
+  fn doc_for_maybe_node(
+    &self,
+    module_symbol: ModuleSymbolRef,
+    symbol: &Symbol,
+    maybe_node: Option<SymbolNodeRef<'_>>,
+  ) -> Option<DocNode> {
+    match module_symbol {
+      ModuleSymbolRef::Json(module_symbol) => parse_json_module_doc_node(
+        module_symbol.specifier(),
+        module_symbol.text_info().text_str(),
+      ),
```

Comment:
I don't understand this bit - how can JSON have documentation?

Reply:
JSON modules can be imported then exported as an object. I see there's no test for this so I'll add one.

Reply:
Oh, nevermind. There is a test.